test/kmerge-test.hs

```diff
@@ -66,12 +69,12 @@ main = do
                     -- (because the input values are unformly random,
                     -- there shouldn't be a lot of "cheap" leftovers elements,
                     -- i.e. when other inputs are exhausted, but there are few)
-                    [ testCount "sortConcat"      3190 (L.sort . concat) input8
-                    , testCount "listMerge"       3479 listMerge         input8
-                    , testCount "treeMerge"       2391 treeMerge         input8
-                    , testCount "heapMerge"       3168 heapMerge         input8
-                    , testCount "loserTreeMerge"  2391 loserTreeMerge    input8
-                    , testCount "mutHeapMerge"    3169 mutHeapMerge      input8
+                    [ testCount "sortConcat"      3185 (L.sort . concat) input8
```

Comment:
I don't follow why changing the key type changes the number of comparisons. Do you understand what's going on with the change in counts?

Or is it because the random generator is different for the different representation?

Reply:
Exactly that. It's because with the changes here we only generate the last 8 bytes of the keys randomly, the rest is a fixed prefix.